src/components/pages/Verify/index.tsx

```diff
@@ -41,56 +41,127 @@ type OwnProps = NavigationScreenProps<{ toMemberId: MemberId }>;
 
 type VerifyProps = OwnProps & ReduxStateProps;
 
+interface BaseVerifyState {
+  // necessary for handling back button presses.
+  previousSteps: CurrentVerifyState[];
+}
+
+/**
+ * Current state of the verification flow.
+ *
+ * There are two flows. In either case, you are logged in and verifying someone
+ * else. They are:
+ *
+ * ## You invited someone to Raha, and therefore a video of you verifying the
+ *    other person should already exist
+ *
+ * 1. CONFIRM_EXISTING_VIDEO. Video is present, you can confirm it or retake. If
+ *    you confirm it:
+ * 2. CONFIRM_VERIFICATION: All necessary information is present. Complete the
+ *    verification process.
+ *
+ * Else, go to the CAMERA step, and follow the flow as though you don't have an
+ * existent video.
+ *
+ * ## You are verifying someone you didn't invite, and therefore a video of you
+ *    verifying the other person does not exist.
+ *
+ * 1. SPLASH: Informational page
+ * 2. CAMERA: Take a video verifying the other person.
+ * 3. CONFIRM_RECORDED_VIDEO: Confirm that the video is acceptable, or retake
+ *    (go back to CAMERA step). If it's good, then proceeding to the next
+ *    step uploads it.
+ * 4. CONFIRM_VERIFICATION: See above.
+ */
+type CurrentVerifyState =
```

Comment:
nit: maybe name `VerifyStepState` and have the `VerifyStep` state members be `previousVerifyStepState` and `currentVerifyStepState`? I find that more clear than overloading `State`.